### ISSUE TITLE
create_network failed when passing driver overlay

### ISSUE THREAD

**Issue by yaakov-berkovitch** [docker/docker-py#1323]
I am running version 1.12.3 experimental and docker.py API version 1.24.
I want to create an overlay network using the client API create_network, and it failed with the following stack error:
```
Traceback (most recent call last):
  File "/home/paas/tools/paas-orchestrator/org/openpaas/orchestrator/docker/dockerClient.py", line 120, in dockerCreateNetworks
    **client.create_network(network, driver="overlay")**
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/utils/decorators.py", line 35, in wrapper
    return f(self, *args, **kwargs)
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/api/network.py", line 122, in create_network
    return self._result(res, json=True)
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/api/client.py", line 214, in _result
    self._raise_for_status(response)
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/api/client.py", line 210, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/errors.py", line 28, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.APIError: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.24/networks/create ("rpc error: code = 3 desc = **driver name: if driver is specified name is required**")

```
The error reported is not clear and looks not related - I provided the driver name in the call.

Thanks.

**Comment by shin-** [comment]
Can you share an excerpt of the code you're running? Also, what version of docker-py are you running?

**Comment by yaakov-berkovitch** [comment]
docker-py version is 2.0.0-dev.
I extracted the call in a very simple code, and it occurs again.

Following the code that fails:

```
import os,sys
import docker
from docker import APIClient

client = APIClient(base_url='unix://var/run/docker.sock')

newNetwork = client.create_network('my_network', driver="overlay")
print('Network created ' + str(newNetwork))

Traceback (most recent call last):
  File "docker-create-service.py", line 7, in <module>
    newNetwork = client.create_network('my_network', driver="overlay")
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/utils/decorators.py", line 35, in wrapper
    return f(self, *args, **kwargs)
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/api/network.py", line 122, in create_network
    return self._result(res, json=True)
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/api/client.py", line 214, in _result
    self._raise_for_status(response)
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/api/client.py", line 210, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/home/paas/tools/paas-orchestrator/ThirdPartyPackages/docker/errors.py", line 28, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.APIError: 500 Server Error: Internal Server Error for url: http+docker://localunixsocket/v1.24/networks/create ("rpc error: code = 3 desc = driver name: if driver is specified name is required")

```
Thanks.

**Comment by yaakov-berkovitch** [comment]
i closed it by error...so i reopened it.

**Comment by yaakov-berkovitch** [comment]
This issue can be closed because it was related to docker 1.12 experimental. Bug in the REST API.
